src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -289,6 +295,55 @@ public void processArguments() {
         if (!cli.isBlank() && !loaded.isEmpty()) {
             uiCommands.add(new UiCommand.OpenDatabases(loaded));
         }
+
+        if (cli.isCheckConsistency()) {
+            checkConsistency(cliPreferences, entryTypesManager);
+        }
+    }
+
+    private void checkConsistency(CliPreferences cliPreferences,
+                                  BibEntryTypesManager entryTypesManager) {
+        Optional<String> fileName = Optional.ofNullable(cli.getCheckConsistency());
+
+        if (fileName.isEmpty()) {
+            System.out.println(Localization.lang("No file specified for consistency check."));
+            return;
+        }
+
+        Optional<String> outputFormat = Optional.ofNullable(cli.getCheckConsistencyOutputFormat());
+
+        try {
+            Path filePath = Path.of(fileName.get());
+            ParserResult pr = OpenDatabase.loadDatabase(filePath, cliPreferences.getImportFormatPreferences(), fileUpdateMonitor);
+            BibDatabaseContext databaseContext = pr.getDatabaseContext();
+            List<BibEntry> entries = databaseContext.getDatabase().getEntries();
+
+            BibliographyConsistencyCheck consistencyCheck = new BibliographyConsistencyCheck();
+            BibliographyConsistencyCheck.Result result = consistencyCheck.check(entries);
+
+            if (outputFormat.isEmpty()) {
+                Writer ouputStreamWriter = new OutputStreamWriter(System.out);
+                BibliographyConsistencyCheckResultTxtWriter txtWriter = new BibliographyConsistencyCheckResultTxtWriter(result, ouputStreamWriter, entryTypesManager, databaseContext.getMode());
+
+                txtWriter.writeFindings();
+                return;
+            }
+
+            String outputFileName = fileName.get().substring(0, fileName.get().lastIndexOf('.')) + "_consistency_check." + outputFormat.get().toLowerCase();
+
+            try (Writer writer = new FileWriter(outputFileName, StandardCharsets.UTF_8)) {
+                if ("csv".equalsIgnoreCase(outputFormat.get())) {
+                    BibliographyConsistencyCheckResultCsvWriter csvWriter = new BibliographyConsistencyCheckResultCsvWriter(result, writer, entryTypesManager, databaseContext.getMode());
```

Comment:
Better wrap this in try-with-resources; not sure what happens with the `writer`; if there are troubles, please open the writers both in the try-with-resources (and accept duplication of writer creation)